### PR TITLE
New avg filter and bmp085 update

### PR DESCRIPTION
**Pull request by lucasdemarchi**
This is a rebase of an old PR of mine: #2555

I'm resubmitting it because:
 - The same driver could be used for BMP280 and this is the driver available in qflight.
 - The new average filter is better than the current one and useful for other places of the codebase

Just a note that I'm not actually using this sensor anymore and don't plan to use in near future.... but I like to improve what I can in it.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
Updates:
  - use different conversion times for different oversampling configurations
  - add documentation to the new filter
  - s/_count/_has_sample/ and make it bool since now it's just a flag and not a count anymore

**Comment by lucasdemarchi** [comment]
@guludo ^

**Comment by guludo** [comment]
Next feature GitHub could add: show the PR's HEAD history.

File: libraries/AP_Baro/AP_Baro_BMP085.cpp [hunk 1]
```diff
@@ -237,5 +240,23 @@ bool AP_Baro_BMP085::_data_ready()
         return AP_HAL::millis() > _last_temp_read_command_time + 5;
     }
 
-    return AP_HAL::millis() > _last_press_read_command_time + 26;
+    uint32_t conversion_time_msec;
```
Comment (guludo) [inline libraries/AP_Baro/AP_Baro_BMP085.cpp]: What about making this static?

**Comment by guludo** [comment]
@lucasdemarchi Thanks for the changes. There's just one minor inline comment, but it looks good to me as is.

File: libraries/AP_Baro/AP_Baro_BMP085.cpp (hunk 1 above)
Comment (lucasdemarchi) [inline libraries/AP_Baro/AP_Baro_BMP085.cpp]: OVERSAMPLING is known at compile time, which means the compiler will resolve the switch below at compile time.

**Comment by lucasdemarchi** [comment]
Applied, thanks for reviewing it.

**Pull request closed without merging**